### PR TITLE
add update_fahrstrasse to RouteController

### LINKED ISSUE #352: add ```update_fahrstrasse``` to RouteController
## User stories
As a developer I want to call a method when a train enters a platform in order to set the next route for this train brfor it reaches the another track, that may not be part of it's route.

## Acceptance Criteria
- [x] ```update_fahrstrasse``` finds the route to the next platform
- [x] it sets the SUMO route
- [x] it sets the interlocking route

### PR DESCRIPTION
**Pull request by Lietze**
Fixes #352 

This is not testable until there is a functioning PlanPro and Sumo network.

## PR checklist

- [x] Acceptance criteria fulfilled
- [ ] Additional features are tested
- [x] Docs (code, wiki & diagrams) updated
- [x] Breaking changes anounced
- [x] Added classes that inherit from `BaseModel` to `src.constants.tables`
- [ ] Dev-branch has been merged into local branch to resolve conflicts
- [ ] Tests and linter have passed AFTER local merge
- [ ] Another dev reviewed and approved

### CODE REVIEW

File: src/interlocking_component/route_controller.py [hunk 1]
```diff
@@ -190,16 +182,47 @@ def maybe_update_fahrstrasse(self, train: Train, track: Track):
...
+
+        :param track: The track to which the route is searched
+        :type track: Track
+        :return: The interlocking Route corresponding to the track
+        :rtype: Route
+        """
+        for route_candidate in self.interlocking.active_routes:
+            interlocking_track_candidat = route_candidate.contains_segment(
+                track.identifier.split("-re")[0]
+            )
```
Comment (instant-sky) [inline src/interlocking_component/route_controller.py:218]: Is it possible that a track belongs to more than one route?

File: src/interlocking_component/route_controller.py [hunk 2]
```diff
@@ -190,16 +182,47 @@ def maybe_update_fahrstrasse(self, train: Train, track: Track):
...
+    def _free_route(self, track: Track):
+        """This method frees the route corresponding to the given track.
+
+        :param track: One track of the active Route
+        :type track: Track
+        """
+        route = self._get_interlocking_route_for_track(track)
+
+        if route is not None:
+            # This frees the least route in the interlocking
```
Comment (instant-sky) [inline src/interlocking_component/route_controller.py]: typo? least -> last?

**Review by instant-sky — COMMENTED** [review]
Good job, looks very promising! I only have two small questions

File: src/interlocking_component/route_controller.py (hunk 2 above)
Comment (Lietze) [reply src/interlocking_component/route_controller.py]: Yes that is a typo. Thank you!

File: src/interlocking_component/route_controller.py (hunk 1 above)
Comment (Lietze) [reply src/interlocking_component/route_controller.py:218]: It is not possible that a track belongs to more than one **active** route. I will add a comment about this, as this question has come up before.

Comment (instant-sky) [reply src/interlocking_component/route_controller.py:218]: alright, thanks for clarifying!

**Review by instant-sky — APPROVED** [review]

**Pull request merged**